### DISCUSSION TITLE
which parameter is the test file name used by predict.py

### DISCUSSION THREAD

**Discussion by greatofdream** [PaddlePaddle/PaddleSpatial#166] [BaiduKDDCup2022] [answered]
I saw the demo predict.py use `settings["path_to_test_x"]` in class Experiment:get_test_x, is this mean the test file name will set as the value of `settings["path_to_test_x"]` and be used for reading test data?

If above is right, it's very strange to couple this parameter into code deeply.

**Comment by shaido987** [comment, marked as answer]
Yes, `settings["path_to_test_x"]` is the file with the data you have available for the prediction. In the evaluation.py code, all test files are iterated over and the value of `settings["path_to_test_x"]` is changed in each iteration (see [here](https://github.com/PaddlePaddle/PaddleSpatial/blob/a68cbadbfc3d559c4cd39a1d55d52895c286a2be/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation/paddlepaddle/evaluation.py#L129)).

The goal is then to predict the 288 timesteps following the information in this file. 
To access the data, in your predict.py file, you can do something like:

```python
df = pd.read_csv(settings["path_to_test_x"])
```

(note that I'm not involved with the oragnization of the competition.)
